### PR TITLE
Use diagnostichelper and properly pass along severity.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @mavasani

File: src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -57,21 +59,34 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext context)
...
 
-            var workspace = (context.Options as WorkspaceAnalyzerOptions)?.Services.Workspace;
+            var options = context.Options as WorkspaceAnalyzerOptions;
+            var workspace = options?.Services.Workspace;
             if (workspace == null)
             {
                 return;
             }
 
+            var optionSet = options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
```
Comment (mavasani) [inline src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs]: I think this proves definite benefit of having a special analyzer in Roslyn.Diagnostics.Analyzers (roslyn-analyzers repo) which is dedicated analyzer package for Roslyn.sln. Let me file an issue.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs]: agreed :)

Comment (mavasani) [reply src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs]: https://github.com/dotnet/roslyn-analyzers/issues/2508

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**